src/fontcache.cpp

```diff
@@ -162,16 +206,15 @@ static bool IsDefaultFont(const FontCacheSubSetting &setting)
 uint GetFontCacheFontSize(FontSize fs)
 {
 	const FontCacheSubSetting &setting = *GetFontCacheSubSetting(fs);
-	return IsDefaultFont(setting) ? FontCache::GetDefaultFontHeight(fs) : setting.size;
+	return (IsDefaultFont(setting) && setting.size == 0) ? FontCache::GetDefaultFontHeight(fs) : setting.size;
```

Comment:
I would split this ternary.

```suggestion
	if (IsDefaultFont(setting) && setting.size == 0)  return FontCache::GetDefaultFontHeight(fs);
	return setting.size;
```